packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -158,6 +158,90 @@ class _SnackBarActionState extends State<SnackBarAction> {
 /// A SnackBar with an action will not time out when TalkBack or VoiceOver are
 /// enabled. This is controlled by [AccessibilityFeatures.accessibleNavigation].
 ///
+/// {@tool dartpad --template=stateless_widget_scaffold_center}
+///
+/// Here is an example of a [SnackBar] with an [action] button implemented using
+/// [SnackBarAction].
+///
+/// ```dart preamble
+/// class _ShowSnackBarButton extends StatelessWidget {
+///   @override
+///   Widget build(BuildContext context) {
+///     return ElevatedButton(
+///       child: Text("Show Snackbar"),
+///       onPressed: () {
+///         Scaffold.of(context).showSnackBar(
+///           SnackBar(
+///             content: Text("Awesome Snackbar!"),
+///             action: SnackBarAction(
+///               label: "Action",
+///               onPressed: () {
+///                 // Code to execute
+///               }
+///             ),
+///           ),
+///         );
+///       },
+///     );
+///   }
+/// }
```

Comment:
It seems unnecessary to refactor this into its own stateless widget. Couldn't you just put this content into the `build` function? Here and in the next example.

Reply:
I'll refactor the ```_ShowSnackBarButton``` into the build function. For the next example do you want me to remove ```_customSnackBar()``` and place it directly inside the ```showSnackBar()``` call inside ```build()```?

Reply:
Yeah, I think that makes more sense. My rationale is that we usually separate out a new widget class only when it's being reused, and in this case, it isn't since it's the only instance in the example :)

Reply:
Alright. I'm working on it.

Reply:
I have made the required changes. Please review.